### DISCUSSION TITLE
Trace Segments in browser monitoring cannot be associated with error logs

### DISCUSSION THREAD

**Discussion by justwiner** [apache/skywalking#11898] [Q&A] [answered]
Hello everyone, I have a question about the log association of browser monitoring.

I want to query the error log associated with a Trace Segment through the following methods.

In fact, an error log has been generated, but it cannot be queried through this method.

![image](https://github.com/apache/skywalking/assets/30335393/97acb071-5eef-4aa0-805e-9f66c775da36)
![image](https://github.com/apache/skywalking/assets/30335393/9c102d33-d2ad-430a-88b4-872c86673745)
![image](https://github.com/apache/skywalking/assets/30335393/97178e34-04bd-47cc-a700-07d8c492ee6c)

But the error message on the page can be seen in the error log

![image](https://github.com/apache/skywalking/assets/30335393/3ca37021-84e5-4d67-801c-2c44d9387e59)

Is there a way to associate this error log with Trace Segments? Or is the "Related Logs" button in Trace Segments actually associated with other logs?

Thank you for your answer!

**Comment by wu-sheng** [comment]
Browser logs are from async way, we don't have a way to associate them with trace context. Trace starting from browser is only HTTP related, but logs are more about executions in the browser.

**Reply by justwiner** [reply]
Thank you for the answer. I have one more question.😊

What logs are associated with the Trace Segment? Because I found a "Related Log" button.

**Reply by wu-sheng** [reply, marked as answer]
Usually, application logs, such as in Java, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-log4j-1.x/
